serversocket.js

```diff
@@ -1427,10 +1427,19 @@ AGServerSocket.prototype.deauthenticate = async function (options) {
 };
 
 AGServerSocket.prototype.kickOut = function (channel, message) {
-  delete this.channelSubscriptions[channel];
-  this.channelSubscriptionsCount--;
-  this.transmit('#kickOut', {channel, message});
-  return this.server.brokerEngine.unsubscribeSocket(this, channel);
+  let channels = channel;
+  if (!channels) {
+    channels = Object.keys(this.channelSubscriptions);
+  }
+  if (!Array.isArray(channels)) {
+    channels = [channel];
+  }
+  for (const channelName of channels) {
+    delete this.channelSubscriptions[channelName];
+    this.channelSubscriptionsCount--;
+    this.transmit('#kickOut', {channel: channelName, message});
+    this.server.brokerEngine.unsubscribeSocket(this, channelName);
```

Comment:
Ideally, the kickOut method should be async and it should await on the `this.server.brokerEngine.unsubscribeSocket(this, channelName);` line. In this case, we would need to use `await Promise.all(...)` instead of the for loop so that channels can be processed in parallel.

Although it doesn't matter for the default brokerEngine since the unsubscribeSocket method is async.
It's in case the user provides a custom brokerEngine - Then it's possible that the unsubscribe could have a delay (not be instant like the default).

Reply:
Good point. I'll try to implement that.

Reply:
Custom brokers could not only introduce a delay, but also could throw something.  
I've simulated a error within `AGSimpleBroker.unsubscribeClient` and have found out that it causes crashes on `socket._destroy`, because unlike `socket._subscribeSocket`, `socket._unsubscribe` doesn't account for possibility of broker errors.

I guess it won't hurt to also emit these two events on `socket.kickOut` too:  
socket 'unsubscribe'  
agServer 'unsubscription'

Reply:
Nice! I'll try it out and aim to get this merged ASAP.